### PR TITLE
docs: add connector reference (en-US)

### PR DESCRIPTION
**Pull request by darcyYe**

### CODE REVIEW

File: docs/docs/references/connectors/README.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import ConnectorStorageIntro from './fragments/_connector_storage_intro.mdx';
+import ConnectorTypeIntro from './fragments/_connector_type_intro.mdx';
+
+# 🔌 Connectors
+
+## Introduction
+
+### What is a connector?
+
+_Connectors_ play a critical role in Logto. With their help, Logto enables end-users to use SMS-based or email-based passwordless registration and the capabilities of signing in with social accounts. With the increasing popularity of websites and applications, passwordless and social sign-ins allow users to avoid managing numerous accounts and passwords.
```
Comment (simeng-li) [inline docs/docs/references/connectors/README.mdx:11]: `enables end-users to use SMS-based or email-based ...` sounds not appropriate.

File: docs/docs/references/connectors/README.mdx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+_Connectors_ are stored both locally and remotely:
+
+<ConnectorStorageIntro />
+
+## Connector's local storage: _ConnectorMetadata_
+
+### id
+
+_**id**_ is an _unique_ string-typed key to identify connector in Logto.
```
Comment (simeng-li) [inline docs/docs/references/connectors/README.mdx:25]:
```suggestion
_**id**_ is an _unique_ string-typed key to identify a connector in Logto.
```

File: docs/docs/references/connectors/README.mdx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+### platform
+
+_**platform**_ should be one of the following values:
+
+<ConnectorPlatformIntro />
+
+### name
+
+_**name**_ is a Javascript/Typescript object whose keys are i18n country codes and values are connectors' display name.
```
Comment (simeng-li) [inline docs/docs/references/connectors/README.mdx:50]: maybe just object?

File: docs/docs/references/connectors/README.mdx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+### logo
+
+_**logo**_ is an URL or relative path of connector's logo (light mode by default).
```
Comment (simeng-li) [inline docs/docs/references/connectors/README.mdx:54]: light mode by default stands for?

File: docs/docs/references/connectors/README.mdx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+### logoDark
+
+_**logo**_ is an _nullable_ URL or relative path of connector's dark mode logo.
```
Comment (simeng-li) [inline docs/docs/references/connectors/README.mdx:58]: _**logoDark**_ ?

File: docs/docs/references/connectors/README.mdx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+:::note
+Usually, _**logo**s_ always exist, and _**logoDark**s_ only exist for some exceptional cases when the light mode logos may become obscure in dark mode.
```
Comment (simeng-li) [inline docs/docs/references/connectors/README.mdx:61]: I think just mentioning logo is required, logoDark is optional should be good enough. We display logo in both light and dark mode if no logoDark is provided.  

`some exceptional cases when the light mode logos may become obscure in dark mode` s` is objective. All we need to show is we support showing different logos under the different color themes.  Leave all the judgment to the users themselves.

File: docs/docs/references/connectors/README.mdx (hunk 1 above)
Comment (darcyYe) [reply docs/docs/references/connectors/README.mdx]: deleted to make it more clear.

File: docs/docs/references/connectors/README.mdx (hunk 3 above)
Comment (darcyYe) [reply docs/docs/references/connectors/README.mdx]: Updated.

File: docs/docs/references/connectors/README.mdx (hunk 4 above)
Comment (darcyYe) [reply docs/docs/references/connectors/README.mdx]: deleted this to make it more clear and add descriptions in Notes.

File: docs/docs/references/connectors/README.mdx (hunk 6 above)
Comment (darcyYe) [reply docs/docs/references/connectors/README.mdx]: Good call, updated.

File: docs/docs/references/connectors/README.mdx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+### What is a connector?
+
+_Connectors_ play a critical role in Logto. With their help, Logto enables end-users to use passwordless registration or sign-in and the capabilities of signing in with social accounts. With the increasing popularity of websites and applications, passwordless and social sign-ins allow users to avoid managing numerous accounts and passwords.
+
+:hushed: Sounds fascinating, Uh? See [Configure connectors](../../recipes/configure-connectors/) if you want to set up an existing connector. If you can not find the connector you want to set up, you may develop those connectors by following the guides in [Create your connector](../../recipes/create-your-connector/).
+
+## Compositions
+
+_Connectors_ are stored both locally and remotely:
```
Comment (IceHe) [inline docs/docs/references/connectors/README.mdx:17]: Seems that the connector data is divided into two parts, one part is hardcoded in the code, and another part is stored in the DB.

**Review by IceHe — APPROVED** [review]
Overall LGTM

File: docs/docs/references/connectors/README.mdx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+### What is a connector?
+
+_Connectors_ play a critical role in Logto. With their help, Logto enables end-users to use passwordless registration or sign-in and the capabilities of signing in with social accounts. With the increasing popularity of websites and applications, passwordless and social sign-ins allow users to avoid managing numerous accounts and passwords.
+
+:hushed: Sounds fascinating, uh? See "[Configure connectors](../../recipes/configure-connectors/)" if you want to set up an existing connector. If you can not find the connector you want to set up, you may develop those connectors by following the guides in "[Create your connector](../../recipes/create-your-connector/)".
+
+## Compositions
+
+As there are lots of properties in connector data, and to make the data loading and updating more efficient, we store part of connector data which will be modified frequently to DB and the rest of that locally.
```
Comment (IceHe) [inline docs/docs/references/connectors/README.mdx:17]: Hard to read. I think so as well.

image

just suggestion:

```suggestion
There are lots of properties in connector data.
To make the connector data loading and updating more efficient, 
we store the frequently updated data in DB and the rest in the locality.
```

**Review by simeng-li — APPROVED** [review]
lgtm

File: docs/docs/references/connectors/README.mdx (hunk 8 above)
Comment (darcyYe) [reply docs/docs/references/connectors/README.mdx]: Updated, good call.

**Pull request merged**